fabtests/common/shared.c

```diff
@@ -764,6 +764,18 @@ int ft_alloc_active_res(struct fi_info *fi)
 		return ret;
 	}
 
+	if (opts.iface == FI_HMEM_CUDA) {
+		bool value = true;
+		ret = fi_setopt(&ep->fid, FI_OPT_ENDPOINT,
+				FI_OPT_CUDA_API_PERMITTED,
+				&value, sizeof(bool));
+		if (ret) {
+			FT_PRINTERR("setopt FI_OPT_CUDA_API_PERMITTED "
+				    "to true failed!", ret);
+			return ret;
+		}
+	}
```

Comment:
I don't like that every fabtest using CUDA now includes this.  Without any sort of comment, it makes it look like it's a required setting for using CUDA.  But all it's trying to do is verify that the provider handles this setopt.  I'd prefer to see this moved into a specific fabtest and clarify that it's only validating support for the setopt, and not some other requirement on using CUDA.

Reply:
In new revision, I added new unit test `fi_setopt_test` to test this option.